src/software/ai/navigator/path_planner/hrvo/brute_force_nearest_neighbor_search.hpp

```diff
@@ -0,0 +1,31 @@
+#pragma once
+
+#include <vector>
+
+/**
+ * Finds the neighbors of a candidate based on a comparator and radius
+ *
+ * @tparam T generic type for the candidate and its neighbors
+ * @tparam Func type for the comparator
+ * @param candidate cannot be null
+ * @param neighbors a list of T objects neighbors, which may contain the candidate object
+ * itself
+ * @param radius the maximum distance from a candidate object
+ * @param comparator function that measures the square distance between a candidate and a
+ * neighbor
+ * @return a list representing a subset of neighbors
+ */
+template <class T, typename Func>
+std::vector<T> nearestNeighbours(const T& candidate, const std::vector<T>& neighbors,
+                                 double radius, Func comparator)
```

Comment:
we can probably rename `comparator` to something like `distanceFunc` so its a bit more clear

Reply:
Done